openmp/libomptarget/plugins-nextgen/common/PluginInterface/PluginInterface.h

```diff
@@ -671,6 +671,20 @@ struct GenericDeviceTy : public DeviceAllocatorTy {
   Error synchronize(__tgt_async_info *AsyncInfo);
   virtual Error synchronizeImpl(__tgt_async_info &AsyncInfo) = 0;
 
+  /// Invokes any global constructors on the device if present and is required
+  /// by the target.
+  virtual Error callGlobalConstructors(GenericPluginTy &Plugin,
+                                       DeviceImageTy &Image) {
+    return Error::success();
```

Comment:
Plugin::success()

Reply:
This code is in the header above the definition of the `Plugin` class, so we can't use that without a complete reordering.